src/api/services/enrichment/enrichment.js

```diff
@@ -0,0 +1,86 @@
+import * as fs from 'fs';
+import path from 'path';
+
+export const createEnrichmentRule = async ctx => {
+    const enrichment = ctx.request.body;
+    if (enrichment.hasOwnProperty('rule')) {
+        return enrichment;
+    }
+
+    if (!enrichment.webServiceUrl || !enrichment.sourceColumn) {
+        throw new Error(`Missing parameters`);
+    }
+
+    try {
+        const excerpt = await ctx.dataset.getExcerpt();
+        const sourceData = excerpt[0][enrichment.sourceColumn];
+        let data;
+        try {
+            data = JSON.parse(sourceData);
+        } catch {
+            data = sourceData;
+        }
+
+        let rule = getEnrichmentRuleModel(data, enrichment);
+
+        return {
+            name: enrichment.name,
+            advancedMode: enrichment.advancedMode,
+            rule: rule,
+        };
+    } catch (e) {
+        console.log('Error:', e.stack);
+    }
+};
+
+export const getEnrichmentRuleModel = (sourceData, enrichment) => {
+    try {
+        let rule;
+        if (typeof sourceData === 'string' || Array.isArray(sourceData)) {
+            const file = Array.isArray(sourceData)
+                ? './directPathMultipleValues.txt'
+                : './directPathSingleValue.txt';
+            rule = fs.readFileSync(path.resolve(__dirname, file)).toString();
+            let columnName = `${enrichment.sourceColumn}${
+                !!enrichment.subPath ? '.' + enrichment.subPath : ''
+            }`;
+            rule = rule.replace(/\[\[COLUMN NAME\]\]/g, columnName);
+            rule = rule.replace(
+                '[[WEB SERVICE URL]]',
+                enrichment.webServiceUrl,
+            );
+        }
+
+        if (typeof sourceData === 'object') {
+            if (!enrichment.subPath) {
+                throw new Error(`Missing sub-path parameters`);
+            }
+            const subPathData = sourceData[0][enrichment.subPath];
+            if (!!!subPathData) {
```

Comment:
`!!!` ? Strange, I've never used this, is it a known pattern to check if the data is false?